src/app/core/auth-service/auth-service.service.ts

```diff
@@ -1,8 +1,135 @@
 import { Injectable } from '@angular/core';
-
+import { Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { Subject } from 'rxjs/Subject';
+import * as auth0 from 'auth0-js';
+//--------------------------------------------------------------------------------
+const rootRole = 'root';
+const investorRole = 'investor';
+const userRole = 'user'; 
+//--------------------------------------------------------------------------------
 @Injectable()
-export class AuthServiceService {
+export class AuthService {
 
-  constructor() { }
+  private eventEmmiter: Subject<string> = new Subject<string>();
+  private name: string = '';
```

Comment:
You don't have to set type explicit types, because it will be implicitly set by TS